bin/node/runtime/src/lib.rs

```diff
@@ -1835,6 +1835,15 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl pallet_nomination_pools_rpc_runtime_api::NominationPoolsApi<Block, AccountId, Balance> for Runtime {
+		fn pending_rewards(member_account: AccountId) -> Balance {
+			if let Some(member) = pallet_nomination_pools::PoolMembers::<Runtime>::get(member_account) {
+				return member.pending_rewards(member.last_recorded_reward_counter).unwrap_or_default();
```

Comment:
Not quite accurate: you must read the reward counter from the reward pool to which this member belongs, as such: 

https://github.com/paritytech/substrate/blob/6784824595bbd038ec971723fc1dd1c5e761ccf3/frame/nomination-pools/src/lib.rs#L2302

Reply:
What you are doing in the companion seems correct.